File: medium/cpp/1983.cpp
// https://leetcode.com/problems/widest-pair-of-indices-with-equal-range-sum/

class Solution
{
public:
    int widestPairOfIndices(std::vector<int>& nums1, std::vector<int>& nums2)
    {
        std::unordered_map<int, int> pref;
        pref[0] = -1;
        auto res = 0;

        for (auto i = 0, s = 0; i < std::size(nums1); i++)
        {
            s += nums1[i] - nums2[i];

            if (const auto it = pref.find(s); it != std::end(pref))
            {
                res = std::max(res, i - it->second);
            }
            else
            {
                pref.emplace(s, i);
            }
        }

        return res;
    }
};

File: medium/python/1983.py
// https://leetcode.com/problems/widest-pair-of-indices-with-equal-range-sum/

class Solution:
    def widestPairOfIndices(self, nums1: List[int], nums2: List[int]) -> int:
        pref = { 0: -1 }
        res = 0
        sum = 0

        for i in range(0, len(nums1)):
            sum += nums1[i] - nums2[i]

            if sum in pref:
                res = max(res, i - pref[sum])
            else:
                pref[sum] = i

        return res
        